afkak/consumer.py

```diff
@@ -594,7 +594,10 @@ def _handle_offset_response(self, responses):
             # Make sure we got a valid offset back. Kafka uses -1 to indicate
             # no committed offset was retrieved
             if response.offset == OFFSET_NOT_COMMITTED:
-                self._fetch_offset = OFFSET_EARLIEST
+                if self.auto_offset_reset in [None, OFFSET_EARLIEST]:
+                    self._fetch_offset = OFFSET_EARLIEST
+                elif self.auto_offset_reset in [OFFSET_LATEST]:
```

Comment:
I'd prefer use of `==` rather than `in` to avoid the temporary list.

Reply:
Done and restructured the conditional check.